pkg/client/metrics/metrics.go

```diff
@@ -23,25 +23,27 @@ import (
 
 	"github.com/prometheus/client_golang/prometheus"
 
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
 	"sigs.k8s.io/prometheus-adapter/pkg/client"
 )
 
 var (
 	// queryLatency is the total latency of any query going through the
 	// various endpoints (query, range-query, series).  It includes some deserialization
 	// overhead and HTTP overhead.
-	queryLatency = prometheus.NewHistogramVec(
-		prometheus.HistogramOpts{
+	queryLatency = metrics.NewHistogramVec(
+		&metrics.HistogramOpts{
 			Name:    "cmgateway_prometheus_query_latency_seconds",
```

Comment:
Since this metric was never expose, we can safely change its name without breaking anyone. I would suggest the following one:

```suggestion
			Namespace:    "prometheus_adapter",
			Subsystem:    "prometheus_client",
			Name:    "request_duration_seconds",
```